data/src/cache.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * EJS_Cache
+ * Manages a cache of files using IndexedDB for storage.
+ */
+class EJS_Cache {
+    /**
+     * Creates an instance of EJS_Cache.
+     * @param {boolean} enabled - Whether caching is enabled.
+     * @param {EJS_STORAGE} storage - Instance of EJS_STORAGE for IndexedDB operations.
+     * @param {number} maxSizeMB - Maximum size of the cache in megabytes.
+     * @param {number} maxAgeMins - Maximum age of items (in minutes) before they are cleaned up.
+     */
+    constructor(enabled = true, storage, maxSizeMB = 4096, maxAgeMins = 7200) {
+        this.enabled = enabled;
+        this.storage = storage;
+        this.maxSizeMB = maxSizeMB;
+        this.maxAgeMins = maxAgeMins;
+        this.minAgeMins = Math.max(60, maxAgeMins * 0.1); // Minimum 1 hour, or 10% of max age
+
+        console.log('Initialized EJS_Cache with settings:', {
+            enabled: this.enabled,
+            storage: this.storage,
+            enabledValue: enabled,
+            maxSizeMB: this.maxSizeMB,
+            maxAgeMins: this.maxAgeMins,
+            minAgeMins: this.minAgeMins
+        });
+    }
+
+    /**
+     * Retrieves an item from the cache.
+     * @param {*} key - The unique key identifying the cached item.
+     * @returns {Promise<EJS_CacheItem|null>} - The cached item or null if not found.
+     */
+    async get(key) {
+        if (!this.enabled) return null;
+
+        const item = await this.storage.get(key);
+        // if the item exists, update its lastAccessed time and return cache item
+        if (item) {
+            item.lastAccessed = Date.now();
+            await this.storage.put(key, item);
+        }
+
+        return item ? new EJS_CacheItem(item.key, item.files, item.added, item.type, item.filename) : null;
+    }
+
+    /**
+     * Stores an item in the cache.
+     * @param {EJS_CacheItem} item - The cache item to store.
+     */
+    async put(item) {
+        if (!this.enabled) return;
+
+        // before putting, ensure item is of type EJS_CacheItem
+        if (!(item instanceof EJS_CacheItem)) {
+            throw new Error("Item must be an instance of EJS_CacheItem");
+        }
+
+        // check if the item exists, if so remove the existing item
+        const existingItem = await this.get(item.key);
+        if (existingItem) {
+            await this.storage.remove(item.key);
+        }
+
+        // check that the size of item.files does not cause the cache to exceed maxSizeMB
+        let currentSize = 0;
+        const allItems = await this.storage.getAll();
```

Comment:
Does this load the entire database into RAM or do you know if the browser only reads it on an attempt to access it?